### PR TITLE
added support for all php versions

### PR DESCRIPTION
**Pull request by vmsachin**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -8,16 +8,25 @@ ns ?= "test"
 
 # Determine the operating system
 UNAME_S := $(shell uname -s)
+PHP_API_VERSION := $(shell php -i | grep 'PHP API' | awk -F' => ' '{print $$2}')
```
Comment (khaf) [inline Makefile:11]: This line is not working for me on Ubuntu 22.04. This regexp should be a more reliable substitute:
`php -i | grep -Po '(?<=PHP API => ).*'`

File: Makefile [hunk 2]
```diff
@@ -8,16 +8,25 @@ ns ?= "test"
...
+PHP_VERSION := $(shell php -v | head -n 1 | awk '{print $$2}' | cut -d. -f1,2)
```
Comment (khaf) [inline Makefile:12]: Same here. This seems to be a better replacement:
`php -i | grep -Po '(?<=PHP Version => ).*' | uniq`

File: Makefile [hunk 3]
```diff
@@ -8,16 +8,25 @@ ns ?= "test"
...
+    $(error PHP version must be greater than or equal to 8.0)
+endif
+
+
 ifeq ($(UNAME_S),Darwin)
     EXTENSION := .dylib
-	EXT_DIR_PATH := /opt/homebrew/opt/php@8.1/pecl/20210902
-    PHP_INI_PATH := /opt/homebrew/etc/php/8.1/php.ini
-    RESTART_COMMAND := brew services restart php@8.1
+	EXT_DIR_PATH := /opt/homebrew/opt/php@$(PHP_VERSION)/pecl/$(PHP_API_VERSION)
```
Comment (khaf) [inline Makefile:22]: Are you assuming that PHP is preinstalled in this path? Do we have enough documentation to assume this dependency will be met on install?

File: Makefile [hunk 4]
```diff
@@ -8,16 +8,25 @@ ns ?= "test"
...
+    PHP_INI_PATH := /opt/homebrew/etc/php/$(PHP_VERSION)/php.ini
+    RESTART_COMMAND := brew services restart php@$(PHP_VERSION)
 else ifeq ($(UNAME_S),Linux)
     EXTENSION := .so
-	EXT_DIR_PATH := /usr/lib/php/20210902/
-    PHP_INI_PATH := /etc/php/8.1/cli/php.ini
-    RESTART_COMMAND := sudo systemctl restart php8.1-fpm && sudo systemctl restart apache2
+	EXT_DIR_PATH := /usr/lib/php/$(PHP_API_VERSION)/
```
Comment (khaf) [inline Makefile:27]: Same here. Ubuntu/Debian/Fedora/Arch may have different installation paths. Have you checked them all?

**Review by khaf — CHANGES_REQUESTED** [review]

File: Makefile (hunk 3 above)
Comment (vmsachin) [reply Makefile:22]: yes for macOS this is the default installation path. There is documentation on homebrew as well confirming this.

File: Makefile (hunk 4 above)
Comment (thias) [reply Makefile:27]: The most reliable is to use `php-config --extension-dir`, which requires php-devel on RHEL/Fedora, and similar devel-only packages on others. A fallback using only php cli is `php -r 'echo ini_get("extension_dir");'`

Comment (vmsachin) [reply Makefile]: Thanks @thias this works!

**Review by khaf — APPROVED** [review]

**Pull request merged**